src/compiler/scala/tools/nsc/typechecker/Macros.scala

```diff
@@ -282,6 +282,10 @@ trait Macros extends MacroRuntimes with Traces with Helpers {
     macroImplBindingCache.getOrElseUpdate(macroDef,
       macroDef.getAnnotation(MacroImplAnnotation) collect {
         case AnnotationInfo(_, List(pickle), _) => MacroImplBinding.unpickle(pickle)
+      } orElse {
+        macroDef.getAnnotation(MacroImplLocationAnnotation) collect {
+          case AnnotationInfo(_, List(unpickled), _) => MacroImplBinding.unpickle(MacroImplBinding.pickle(unpickled))
```

Comment:
`MacroImplBinding.unpickle(MacroImplBinding.pickle(unpickled))` != `unpickled`?

Reply:
not really, what I attach to the `MacroImplLocationAnnotation` is just the tree on the `rhs` of a macro method, without any verification and transformation like is done by `MacroImplBinding.pickle`. But that method can be modified so it can directly go to `MacroImplBinding` data structure rather than this indirection.

Essentially, the reason tasty reader doesn't call `MacroImplBinding.pickle` directly is because it only has access to the `SymbolTable` and not the whole `Global`